### PR TITLE
Add connect by hostname to agent configmap

### PR DESCRIPTION
**Pull request by tuxotron**
When setting the kubeletConnectByHostname to "true", the charts update the agent daemonset to define the MY_NODE_NAME environment variable, but it doesn't set the "node_name_env" setting in the WorkloadAttestor:k8s plugin, therefore the agent continues trying to connect to the kubelet using the localhost interface

### CODE REVIEW

File: charts/spire/charts/spire-agent/templates/configmap.yaml [hunk 1]
```diff
@@ -116,6 +116,9 @@ plugins:
         disable_container_selectors: {{ .Values.workloadAttestors.k8s.disableContainerSelectors }}
         use_new_container_locator: {{ .Values.workloadAttestors.k8s.useNewContainerLocator }}
         verbose_container_locator_logs: {{ .Values.workloadAttestors.k8s.verboseContainerLocatorLogs }}
+        {{- if eq (include "spire-agent.connect-by-hostname" .) "true" }}
+        node_name_env: "MY_NODE_NAME"
+        {{- end }}
```
Comment (kfox1111) [inline charts/spire/charts/spire-agent/templates/configmap.yaml:121]: According to https://github.com/spiffe/spire/blob/main/doc/plugin_agent_workloadattestor_k8s.md
node_name_env is defaulted to MY_NODE_NAME when unset.

So, I think this is a  NOOP.

But having it explicit is always nice.

I'm ok with merging this.

Comment (mrsabath) [reply charts/spire/charts/spire-agent/templates/configmap.yaml:121]: I tested this change on Openshift cluster and it works fine.

**Review by mrsabath — APPROVED** [review]
LGTM!

**Review by kfox1111 — APPROVED** [review]
LGTM. Thanks! :)

**Pull request merged**